balloc/balloc.c

```diff
@@ -1401,14 +1406,15 @@ M0_INTERNAL int m0_balloc_load_extents(struct m0_balloc *cb,
 		 (int)group_spare_fragments_get(grp));
 	M0_PRE(m0_mutex_is_locked(bgi_mutex(grp)));
 
-	if (grp->bgi_extents != NULL) {
+	if (grp->bgi_extents_loaded) {
 		M0_LOG(M0_DEBUG, "Already loaded");
 		return M0_RC(0);
 	}
 
 	if (group_fragments_get(grp) +
 	    group_spare_fragments_get(grp) == 0) {
 		M0_LOG(M0_NOTICE, "zero fragments");
+		grp->bgi_extents_loaded = true;
```

Comment:
I can not add comments for unmodified code snippet.
But let me add here.
Please review the following change:
```
diff --git a/balloc/balloc.c b/balloc/balloc.c
index 95cf1a56a..9647dd282 100644
--- a/balloc/balloc.c
+++ b/balloc/balloc.c
@@ -1434,6 +1434,7 @@ M0_INTERNAL int m0_balloc_load_extents(struct m0_balloc *cb,
        grp->bgi_spare.bzp_freeblocks = 0;
        for (i = 0; i < group_fragments_get(grp) +
             group_spare_fragments_get(grp); i++, ex++) {
+               ex->le_is_alloc = false;
                ex->le_ext.e_end = next_key;
                key = M0_BUF_INIT_PTR(&ex->le_ext.e_end);
                val = M0_BUF_INIT_PTR(&ex->le_ext.e_start);

```
This is to initialize the `le_is_alloc` explicitly (although the default value is false).

Reply:
M0_ALLOC_ARR() -> m0_alloc() -> m0_arch_allocated_zero() would be indirectly initializing it to false, but if still required then can add that explicitly too.

Reply:
I know that is an implicit initialization.
It's up to you to decide whether it is needed.